### PR TITLE
Update ci

### PR DESCRIPTION
**Pull request by rbx**
- Update DDS to 3.7.23.
- Update to new MacOS CI labels.

### CODE REVIEW

File: cmake/legacy.cmake [hunk 1]
```diff
@@ -151,7 +151,7 @@ if(ICU_ROOT)
   set(dds_icu_hint "-DDDS_LD_LIBRARY_PATH=${ICU_ROOT}/lib")
 endif()
 list(APPEND packages dds)
-set(dds_version "3.6")
+set(dds_version "3.7.23")
```
Comment (ChristianTackeGSI) [inline cmake/legacy.cmake:154]: 1. Can you please also update `legacy/README.md` with the new version?
2. I don't know, if the status has changed, but once we were told, that DDS should ship "stable versions". So this has changed?

Comment (rbx) [reply cmake/legacy.cmake:154]: 1. Sure!
2. Don't know, but fine with me if you wanna do that rule. The version which is in dev now didn't compile, so I took the latest. But also I remember @AnarManafov said he will publish stable one this month?

Comment (dennisklein) [reply cmake/legacy.cmake:154]: Yes, he will do within the next week or so. Starting with the next stable release he will change DDS`  release numbering so every every release is considered stable (no distinction any more between even and uneven numbers).

**Comment by dennisklein** [comment]
Can you pls cherry-pick 8eb1f8df2d3692194602ac6dd44818bb67fb6c24 from #516 to fix the brew error?

**Comment by rbx** [comment]
> Can you pls cherry-pick [8eb1f8d](https://github.com/FairRootGroup/FairSoft/commit/8eb1f8df2d3692194602ac6dd44818bb67fb6c24) from #516 to fix the brew error?

done.

**Comment by dennisklein** [comment]
```
SetCTestConfiguration:BuildDirectory:/Users/alfaci/alfaci/workspace/FairRootGroup_FairSoft_PR-518/build
SetCTestConfiguration:SourceDirectory:/Users/alfaci/alfaci/workspace/FairRootGroup_FairSoft_PR-518
SetCTestConfiguration:SubmitURL:https://cdash.gsi.de/submit.php?project=FairSoft
Submit files
   Send to group: Continuous
   SubmitURL: https://cdash.gsi.de/submit.php?project=FairSoft
   Upload file: /Users/alfaci/alfaci/workspace/FairRootGroup_FairSoft_PR-518/build/Testing/20240109-2125/Configure.xml to https://cdash.gsi.de/submit.php?project=FairSoft&FileName=citmac008.gsi.de___PR-518%20macOS%20(legacy)___20240109-2125-Continuous___XML___Configure.xml&build=PR-518%20macOS%20%28legacy%29&site=citmac008.gsi.de&stamp=20240109-2125-Continuous&MD5=9c6f6f2c2acb607ff23228890906f3d9 Size: 7335
   Error when uploading file: /Users/alfaci/alfaci/workspace/FairRootGroup_FairSoft_PR-518/build/Testing/20240109-2125/Configure.xml
   Error message was: LibreSSL/3.3.6: error:0DFFF0A1:lib(13):func(4095):reason(161)
   Problems when submitting via HTTP
CMake Warning at /Users/alfaci/alfaci/workspace/FairRootGroup_FairSoft_PR-518/cmake/FairSoftLib.cmake:183 (message):
   ctest_submit() failed. Continueing
Call Stack (most recent call first):
  /Users/alfaci/alfaci/workspace/FairRootGroup_FairSoft_PR-518/FairSoft_test.cmake:185 (fairsoft_ctest_submit)
```

**Comment by dennisklein** [comment]
https://github.com/Homebrew/homebrew-core/issues/158759

**Comment by dennisklein** [comment]
Based on https://github.com/Homebrew/homebrew-core/issues/158759#issuecomment-1880258090 I tried `brew reinstall -s cmake` but it did not change anything, same error.

**Comment by dennisklein** [comment]
I defined the env variable suggested in https://github.com/Homebrew/homebrew-core/issues/158759#issuecomment-1874091015 (only for citmac008) which fixed the cdash upload.

**Comment by dennisklein** [comment]
> Should I (temporarily?) enable all of these [`master`/Jenkinsfile#L154-L156](https://github.com/FairRootGroup/FairSoft/blob/master/Jenkinsfile?rgh-link-date=2024-01-10T08%3A45%3A50Z#L154-L156) to see how they all handle the update?

Just push your PR branch into the FairRootGroup/FairSoft repo, then it will run everything. (e.g. `git remote add upstream git@github.com:FairRootGroup/FairSoft` and `git push upstream <local-branch-name>:rbx-test`)

**Comment by dennisklein** [comment]
closing this temporarily.

File: cmake/legacy.cmake (hunk 1 above)
Comment (AnarManafov) [reply cmake/legacy.cmake:154]: > 1. Sure!
> 2. Don't know, but fine with me if you wanna do that rule. The version which is in dev now didn't compile, so I took the latest. But also I remember @AnarManafov said he will publish stable one this month?

Hey @rbx ,
yes, sure thing. I am about to release 3.8 either this week or max beg. next. 
I have finished all tests for OS X (intel/arm). Now finishing up the linux part. Once all green, I will push the release.

Comment (dennisklein) [reply cmake/legacy.cmake:154]: @AnarManafov Current `master` branch of DDS **requires** CMake 3.23. FairSoft currently only requires CMake 3.19. Bumping this because of DDS basically means we have to bootstrap a custom CMake on practically every linux distro in our support matrix. Is DDS really using any CMake 3.23 features? Would it be possible to relax this to 3.19?

Comment (AnarManafov) [reply cmake/legacy.cmake:154]: @dennisklein , honestly I don't recall why the cmake version was bumped. I can for sure relax the requirement.
If you have any machine with 3.19, could you try to just cmake DDS, to make sure nothing comes up?

Comment (dennisklein) [reply cmake/legacy.cmake:154]: > If you have any machine with 3.19, could you try to just cmake DDS, to make sure nothing comes up?

Not really, but you may download older pre-compiled CMake versions e.g. here https://cmake.org/files/v3.19/. Just unpack it to a custom directory and prepend this `<custom-dir>/bin` to your `$PATH` temporarily, it will not mess with your system CMake.

Comment (dennisklein) [reply cmake/legacy.cmake:154]: you probably want a minimum of 3.19.5 because of Boost 1.75, see https://gitlab.kitware.com/cmake/cmake/-/commit/82930647607d23b118c84194afc3c70d6d8e79e3.

Comment (AnarManafov) [reply cmake/legacy.cmake:154]: @dennisklein , sounds good. Will do. If works, would 3.19.5 be ok for you as a minimum?

Comment (AnarManafov) [reply cmake/legacy.cmake:154]: @dennisklein , thank you. I have checked 3.19.5. Everything looks ok. As you requested, I relaxed the min cmake version to 3.19. the change is in the master. Once I finish all linx tests, I will push for a tagged release.

**Comment by rbx** [comment]
MacOS 13.6 fails to build ROOT:
```
[ 48%] Built target ClingUtils
[ 48%] Built target Dictgen
[ 48%] Built target rootcling_stage1
[ 48%] Generating G__Core.cxx, ../lib/Core.pcm
While building module 'Core':
While building module 'std' imported from input_line_1:1:
In file included from <module-includes>:17:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/math.h:308:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__type_traits/promote.h:16:
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/cstddef:50:9: error: no member named 'nullptr_t' in the global namespace
using ::nullptr_t;
      ~~^
While building module 'Core':
While building module 'std' imported from input_line_1:1:
In file included from <module-includes>:17:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/math.h:309:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/limits:107:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/type_traits:452:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__type_traits/is_compound.h:14:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__type_traits/is_fundamental.h:14:
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__type_traits/is_null_pointer.h:24:49: error: use of undeclared identifier 'nullptr_t'; did you mean 'nullptr'?
template <>          struct __is_nullptr_t_impl<nullptr_t> : public true_type {};
                                                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__type_traits/is_null_pointer.h:24:49: error: template argument for template type parameter must be a type
template <>          struct __is_nullptr_t_impl<nullptr_t> : public true_type {};
                                                ^~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__type_traits/is_null_pointer.h:23:17: note: template parameter is declared here
template <class _Tp> struct __is_nullptr_t_impl       : public false_type {};
                ^
While building module 'Core':
While building module 'std' imported from input_line_1:1:
In file included from <module-includes>:34:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/complex.h:28:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/ccomplex:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/complex:238:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/stdexcept:47:
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/exception:149:45: error: field has incomplete type 'std::exception_ptr'
    _LIBCPP_INLINE_VISIBILITY exception_ptr(nullptr_t) _NOEXCEPT : __ptr_() {}
                                            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/exception:144:24: note: definition of 'std::exception_ptr' is not complete until the closing '}'
class _LIBCPP_TYPE_VIS exception_ptr
                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/exception:149:5: error: '__abi_tag__' attribute only applies to structs, variables, functions, and namespaces
    _LIBCPP_INLINE_VISIBILITY exception_ptr(nullptr_t) _NOEXCEPT : __ptr_() {}
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__config:689:37: note: expanded from macro '_LIBCPP_INLINE_VISIBILITY'
#  define _LIBCPP_INLINE_VISIBILITY _LIBCPP_HIDE_FROM_ABI
                                    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/__config:673:26: note: expanded from macro '_LIBCPP_HIDE_FROM_ABI'
          __attribute__((__abi_tag__(_LIBCPP_TOSTRING(_LIBCPP_VERSIONED_IDENTIFIER))))
                         ^
While building module 'Core':
While building module 'std' imported from input_line_1:1:
In file included from <module-includes>:34:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/complex.h:28:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/ccomplex:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/complex:238:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/stdexcept:47:
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/c++/v1/exception:149:55: error: expected ';' at end of declaration list
    _LIBCPP_INLINE_VISIBILITY exception_ptr(nullptr_t) _NOEXCEPT : __ptr_() {}
                                                      ^
While building module 'Core':
While building module 'std' imported from input_line_1:1:
...
```

**Comment by rbx** [comment]
That does look similar. However there they cannot compile anything, while here it seems to fail only for ROOT, while a bunch of stuff before compiles fine.
I could also compile the test code they suggest, if I just login to the node. also if I extend it with nulltr_t:
```
alfaci@citmac001 ~ % cat hello.c 
#include <stdio.h>
#include <cstddef>

using ::nullptr_t;

int main(int argc, char ** argv) {
    printf("Hello Cruel World!\n");
    return 0;
}
alfaci@citmac001 ~ % clang++ -o hello hello.c
clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
alfaci@citmac001 ~ % ./hello 
Hello Cruel World!
```
So somehow the ROOT is mixing things up.

**Comment by dennisklein** [comment]
It sounds like it is just the typical break we get since years once there are new major macos releases, I have summarized the issue here: https://github.com/root-project/cling/issues/457

tldr; The patch I proposed back then was: https://github.com/dennisklein/root/commit/ee75437c09a45b362df0d1c87ea5d9f0dd8123b5.patch?full_index=1 (It was never merged, probably because mac folks just don't care enough, once they updated their own notebook and they are not affected by this any more... ;) )

A worse alternative to the above patch would be to fix the `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk` symlink in the filesystem to point to the sdk version matching the operating system version.

**Comment by dennisklein** [comment]
And note, this issue seems to be present only on **CommandLineTools-only** systems. Full XCode installations appear unaffected by this which is another reason this issue has no priority to non-packaging and non-CI folks.

**Comment by rbx** [comment]
Changing symlink works. Changed it back now to also test your patch. It's in progress: https://alfa-ci.gsi.de/blue/organizations/jenkins/FairRootGroup%2FFairSoft/detail/rbx-update-ci-2/1/pipeline/40

**Comment by dennisklein** [comment]
Ok, looks like it is compiling now. The geant3 segfaults are reported in https://github.com/vmc-project/geant3/issues/42#issuecomment-1867422297 and the [cling + `boost/program_options.hpp` issue](https://github.com/root-project/root/issues/12762) is being addressed in https://github.com/FairRootGroup/FairRoot/pull/1478.

**Comment by dennisklein** [comment]
https://cdash.gsi.de/testDetails.php?test=18537688&build=433018 this may be worth looking into still.

Regarding this PR, I guess we can merge it, however, let's please remove the DDS bump for now and wait until we have the 3.8 release.

**Comment by dennisklein** [comment]
image

not sure, it won't let me re-open. perhaps you have to create a new PR.

**Comment by ChristianTackeGSI** [comment]
> Ok, looks like it is compiling now. The geant3 segfaults are reported in [vmc-project/geant3#42 (comment)](https://github.com/vmc-project/geant3/issues/42#issuecomment-1867422297) and the [cling + `boost/program_options.hpp` issue](https://github.com/root-project/root/issues/12762) is being addressed in [FairRootGroup/FairRoot#1478](https://github.com/FairRootGroup/FairRoot/pull/1478).

Just merged <https://github.com/FairRootGroup/FairRoot/pull/1478>. So that one is removed from this list.

> not sure, it won't let me re-open. perhaps you have to create a new PR.

I had the same issue at some point in time. It basically means to create a new PR…
